dallinger/db.py

```diff
@@ -100,3 +102,26 @@ def wrapper(*args, **kw):
                 else:
                     raise
     return wrapper
+
+
+# Reset outbox when session begins
+@event.listens_for(Session, 'after_begin')
+def after_begin(session, transaction, connection):
+    session.info['outbox'] = []
+
+
+# Reset outbox after rollback
+@event.listens_for(Session, 'after_soft_rollback')
+def after_soft_rollback(session, previous_transaction):
+    session.info['outbox'] = []
+
+
+# Publish messages to redis after commit
+@event.listens_for(Session, 'after_commit')
```

Comment:
This is designed to flush the queue of outgoing messages after any commit, but in practice the create_participant route is the only one that actually adds a message to the queue at this point.